src/main/java/org/openrewrite/staticanalysis/RemoveToStringCallsFromArrayInstances.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.staticanalysis;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+import java.util.List;
+
+public class RemoveToStringCallsFromArrayInstances extends Recipe {
```

Comment:
Could be https://rules.sonarsource.com/java/RSPEC-2116

I would also like to propose that we change the name, since we are either replacing `Object#toString()` with `Arrays.toString()` and sometimes we just add the latter. So maybe something like `UseArraysToStringForArrays`?

Reply:
It is https://rules.sonarsource.com/java/RSPEC-2116. My bad for not tagging it initially. That name change makes sense to me.